client/src/components/ProfilePublic/ProfilePublic.jsx

```diff
@@ -0,0 +1,131 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router-dom';
+
+import { fetchUser, clearUser } from '../../actions/userActions';
+
+import PageContainer from '../Shared/PageContainer/PageContainer';
+import CircleImg from '../Shared/CircleImg/CircleImg';
+import Badge from '../Shared/Badge/Badge';
+
+import { ProfileSection, Header, About, ProfileName } from './ProfilePublic.styled';
+
+class ProfilePublic extends Component {
+	componentDidMount() {
+		console.log('MOUNTED');
+		if (!this.props.isMainUser) {
+			// fetch another users' profile
+			this.props.fetchUser(this.props.match.params.id);
+		}
+	}
+
+	componentDidUpdate(prevProps) {
+		if (this.props.match.params.id !== prevProps.match.params.id) {
+			this.props.fetchUser(this.props.match.params.id);
+		}
+	}
+
+	componentWillUnmount() {
+		this.props.clearUser();
+	}
+
+	render() {
+		const { history, user } = this.props;
+
+		console.log('USER', user);
+
+		return (
+			<PageContainer
+				breadCrumbs={{
+					crumbHistory: [{ name: 'Dashboard', link: '/dashboard' }],
+					current: 'Profile',
```

Comment:
Wasn't sure how to handle this, but that would be a good solution. I recommend that this should be only one level deep because the user can go through profiles indefinitely if the right condition appears and that wouldn't look that great in breadcrumbs.